packages/react/src/components/PhoneNumberInput/PhoneNumberInput.tsx

```diff
@@ -0,0 +1,142 @@
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {FlagOutlined} from '@mui/icons-material';
+import Select, {SelectChangeEvent, SelectProps as MuiSelectProps} from '@mui/material/Select';
+import clsx from 'clsx';
+import {
+  ChangeEvent,
+  forwardRef,
+  ForwardRefExoticComponent,
+  MutableRefObject,
+  ReactElement,
+  useEffect,
+  useState,
+} from 'react';
+import Flag from 'react-world-flags';
+import {countries, Country} from './constants';
+import {WithWrapperProps} from '../../models';
+import {composeComponentDisplayName} from '../../utils';
+import Box, {BoxProps} from '../Box';
+import InputLabel from '../InputLabel';
+import ListItemIcon from '../ListItemIcon';
+import MenuItem from '../MenuItem';
+import './phone-number-input.scss';
+import OutlinedInput, {OutlinedInputProps as MuiOutlinedInputProps} from '../OutlinedInput';
+import Typography from '../Typography';
+
+export interface PhoneNumberInputProps extends BoxProps {
+  OutlinedInputProps?: Omit<MuiOutlinedInputProps, 'id' | 'label' | 'placeholder' | 'value' | 'type'>;
+  SelectProps?: Omit<MuiSelectProps, 'labelId' | 'id' | 'value' | 'onChange' | 'placeholder'>;
+  defaultCountry?: Country;
+  onChange?: (phoneNumber: string) => void;
+  placeholder?: string;
+}
+
+const COMPONENT_NAME: string = 'PhoneNumberInput';
+
+const PhoneNumberInput: ForwardRefExoticComponent<PhoneNumberInputProps> & WithWrapperProps = forwardRef(
+  (props: PhoneNumberInputProps, ref: MutableRefObject<HTMLDivElement>): ReactElement => {
+    const {
+      className,
+      defaultCountry,
+      label,
+      InputLabelProps,
+      OutlinedInputProps,
+      onChange,
+      placeholder,
+      SelectProps,
+      ...rest
+    } = props;
+
+    const classes: string = clsx('oxygen-phone-number-input', className);
+
+    const [country, setCountry] = useState<Country>(defaultCountry ?? countries[0]);
+    const [phoneNumber, setPhoneNumber] = useState<string>('');
+
+    useEffect(() => {
+      if (onChange) {
+        onChange(`${country.dialCode}${phoneNumber}`);
+      }
+    }, [country, phoneNumber, onChange]);
+
+    const handleCountryCodeChange = (event: SelectChangeEvent): void => {
+      setCountry(countries.find((item: Country) => item.dialCode === event.target.value));
+    };
+
+    const handlePhoneNumberChange = (event: ChangeEvent<HTMLInputElement>): void => {
+      setPhoneNumber(event.target.value);
```

Comment:
Shall we remove the useEffect and emit the values on the two onChange event handlers?
```jsx
*interface*
onChange(countryCode, phoneNumber)

...
      setPhoneNumber(event.target.value);
      onChange(country, event.target.value)
```